test/openshift/component.test.ts

```diff
@@ -286,6 +288,40 @@ suite('Openshift/Component', () => {
         });
     });
 
+    suite('linkService', () => {
+        let quickPickStub: sinon.SinonStub;
+
+        setup(() => {
+            quickPickStub = sandbox.stub(vscode.window, 'showQuickPick');
+        });
+
+        test('works from context menu', async () => {
+            quickPickStub.resolves(serviceItem);
+            const result = await Component.linkService(componentItem);
+
+            expect(result).equals(`service '${serviceItem.getName()}' successfully linked with component '${componentItem.getName()}'`);
+            expect(execStub).calledOnceWith(`odo project set ${projectItem.getName()} && odo application set ${appItem.getName()} && odo component set ${componentItem.getName()} && odo link ${serviceItem.getName()} --wait`);
+        });
+
+        test('returns null when no service Link found', async () => {
+            quickPickStub.resolves();
+            const result = await Component.linkService(componentItem);
+
+            expect(result).null;
+        });
+
+        test('errors when a subcommand fails', async () => {
```

Comment:
This test should fail if there was no error triggered. In current implementation it will pass.